tests/baselines/reference/conditionalTypes1.types

```diff
@@ -963,10 +963,10 @@ interface B2 extends A {
 >c : NewDiff<keyof this, "a">
 }
 type c1 = B1['c']; // 'c' | 'b'
->c1 : OldDiff<"a" | "b" | "c", "a">
+>c1 : OldDiff<keyof B1, "a">
 
 type c2 = B2['c']; // 'c' | 'b'
->c2 : OldDiff<"a" | "b" | "c", "a">
+>c2 : "b" | "c"

```

Comment:
It must be because the first (for some meaning of first) definition of the `"b" | "c"` union originated in an `OldDiff` instantiation. That's one of the behaviors this PR gets rid of.